### ISSUE TITLE
Fails to build native image for Scala 3 application which uses kind-projector

### ISSUE THREAD

**Issue by arixmkii** [oracle/graal#4530]
**Describe the issue**
Native image fails for Scala 3 application with kind-projector plugin enabled due to being unable to locate needed reflective data. The problem doesn't persist for Scala 2 applications. [Example project](https://github.com/arixmkii/native-image-issue-scala3)

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. `git clone --depth 1 https://github.com/arixmkii/native-image-issue-scala3.git`
2. Install JVM (any modern 11, 17, 18 or GraalVM will do, build will download graalvm and tools)
3. Install [SBT](https://www.scala-sbt.org/download.html) build tool (standalone, via brew or sdkman, any way for your platform)
4. execute at checkout root `sbt "++3.1.2; nativeImage"`

**Describe GraalVM and your environment:**
 - GraalVM version 22.1.0 for Apple Silicon on macOS
 - JDK major version: 17
 - OS: macOS Monterey
 - Architecture: aarch64 (Apple Silicon)

**More details**
`--verbose` flag added to build didn't give any useful additional output around exception.

Native image command line is handled by [sbt plugin](https://github.com/scalameta/sbt-native-image)

The problem doesn't reproduce for Scala 2 build, which could be performed with `sbt "++2.13.8; nativeImage"`

Reported error for Scala 3.

```
Fatal error: java.lang.TypeNotPresentException: Type io.github.arixmkii.MyTypes$WrappedFImpl.T not present
  | => fat java.base/sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:117)
	at java.base/sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:125)
	at java.base/sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
	at java.base/sun.reflect.generics.visitor.Reifier.reifyTypeArguments(Reifier.java:68)
	at java.base/sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:138)
	at java.base/sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
	at java.base/sun.reflect.generics.repository.MethodRepository.computeReturnType(MethodRepository.java:75)
	at java.base/sun.reflect.generics.repository.MethodRepository.getReturnType(MethodRepository.java:66)
	at java.base/java.lang.reflect.Method.getGenericReturnType(Method.java:295)
	at com.oracle.svm.reflect.hosted.ReflectionDataBuilder.registerTypesForMethod(ReflectionDataBuilder.java:483)
	at com.oracle.svm.reflect.hosted.ReflectionDataBuilder.processMethodMetadata(ReflectionDataBuilder.java:284)
	at com.oracle.svm.reflect.hosted.ReflectionDataBuilder.duringAnalysis(ReflectionDataBuilder.java:185)
	at com.oracle.svm.reflect.hosted.ReflectionFeature.duringAnalysis(ReflectionFeature.java:189)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$10(NativeImageGenerator.java:726)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:74)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$11(NativeImageGenerator.java:726)
	at com.oracle.graal.pointsto.PointsToAnalysis.runAnalysis(PointsToAnalysis.java:751)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:723)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:558)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:515)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:585)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:615)
Caused by: java.lang.ClassNotFoundException: io.github.arixmkii.MyTypes$WrappedFImpl.T
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:587)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at java.base/sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:114)
	... 23 more
```

**Comment by arixmkii** [comment]
Changing main method to
```
  def main(args: Array[String]): Unit = {
    Try(method[WrappedF[Id, *], Unit]).getOrElse(println("Ooops"))
  }
```
Then running with native image agent generated empty `reflect-config.json`.

**Comment by vjovanov** [comment]
I believe this is due to the filter in the agent:
https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/trace/AccessAdvisor.java#L93

I believe we should not filter `sun.reflect.generics.factory` in the agent.

**Comment by arixmkii** [comment]
This specific one seems to be fixed in Scala 3.2.0 nightlies. The demo project builds successfully with scala `3.2.0-RC1-bin-20220523-6783853-NIGHTLY`

Might be worth to keep this open until 3.2.0 stable release for visibility for others encountering the same issue.

**Comment by vjovanov** [comment]
Marking as closed as Scala 3.2 is out.
